app/assets/javascripts/admin/views/user/user_list_view.js

```diff
@@ -1,105 +1,290 @@
+// @flow
+/* eslint-disable jsx-a11y/href-no-hash */
+
 import _ from "lodash";
-import app from "app";
-import Utils from "libs/utils";
-import Marionette from "backbone.marionette";
-import Toast from "libs/toast";
-import SelectAllRows from "libs/behaviors/select_all_rows_behavior";
+import React from "react";
+import { Table, Tag, Icon, Spin, Button, Input } from "antd";
+import Request from "libs/request";
 import TeamRoleModalView from "admin/views/user/team_role_modal_view";
 import ExperienceModalView from "admin/views/user/experience_modal_view";
-import UserListItemView from "admin/views/user/user_list_item_view";
-
-class UserListView extends Marionette.CompositeView {
-  static initClass() {
-    this.prototype.template = _.template(`\
-<h3>Users</h3>
-<table class="table table-striped">
-  <thead>
-    <tr>
-      <th>
-        <input type="checkbox" class="select-all-rows">
-      </th>
-      <th>Last name</th>
-      <th>First name</th>
-      <th>Email</th>
-      <th>Experiences</th>
-      <th>Teams - Role</th>
-      <th class="center-text">Status</th>
-      <th></th>
-    </tr>
-  </thead>
-  <tbody></tbody>
-</table>
-
-<div class="navbar navbar-default navbar-fixed-bottom">
-  <div class="navbar-form">
-    <div class="btn-group">
-      <a class="btn btn-default" id="team-role-modal">
-        <i class="fa fa-group"></i>Edit Teams
-      </a>
-      <a class="btn btn-default" id="experience-modal">
-        <i class="fa fa-trophy"></i>Change Experience
-      </a>
-    </div>
-  </div>
-</div>
-<div id="modal-wrapper"></div>\
-`);
-    this.prototype.className = "user-administration-table container wide";
-    this.prototype.childView = UserListItemView;
-    this.prototype.childViewContainer = "tbody";
-
-    this.prototype.ui = { modalWrapper: "#modal-wrapper" };
-
-    this.prototype.events = {
-      "click #team-role-modal": "showTeamRoleModal",
-      "click #experience-modal": "showExperienceModal",
-    };
+import TemplateHelpers from "libs/template_helpers";
+import Utils from "libs/utils";
+import type { APIUserType, APITeamRole } from "admin/api_flow_types";
 
-    this.prototype.behaviors = {
-      SelectAllRows: {
-        behaviorClass: SelectAllRows,
-      },
-    };
-  }
+const { Column } = Table;
+const { Search } = Input;
 
-  initialize() {
-    this.collection.setPageSize(50);
-    this.collection.fetch({
-      data: "isEditable=true",
-    });
+class UserListView extends React.PureComponent {
+  state: {
+    isLoading: boolean,
+    users: ?Array<APIUserType>,
+    selectedUserIds: Array<string>,
+    isExperienceModalVisible: boolean,
+    isTeamRoleModalVisible: boolean,
+    isActivationFilterSet: ?boolean,
```

Comment:
I see, I made another comment where isActivationFilter was not properly checked. I would advocate for another type, since an optional boolean is a bit weird (false and null are both falsy which makes the ternary distinction error-prone in my opinion). I'd suggest something like `activationFilter: "true" | "false" | null` (Note that the valid filter values are both truthy strings).